ctapipe/image/muon/muon_processor.py

```diff
@@ -68,71 +55,48 @@ class MuonAnalysis(Tool):
-        "i": "EventSource.input_url",
-        "input": "EventSource.input_url",
-        "o": "MuonAnalysis.output",
-        "output": "MuonAnalysis.output",
-        "max-events": "EventSource.max_events",
-        "allowed-tels": "EventSource.allowed_tels",
-    }
-
-    flags = {
-        "overwrite": ({"MuonAnalysis": {"overwrite": True}}, "overwrite output file")
-    }
-
-    def setup(self):
-        if self.output is None:
-            raise ToolConfigurationError("You need to provide an --output file")
-
-        if self.output.exists() and not self.overwrite:
-            raise ToolConfigurationError(
-                "Outputfile {self.output} already exists, use `--overwrite` to overwrite"
-            )
-
-        self.source = EventSource(parent=self)
-        subarray = self.source.subarray
-
-        self.calib = CameraCalibrator(subarray=subarray, parent=self)
+    def __init__(
+        self, subarray: SubarrayDescription, config=None, parent=None, **kwargs
+    ):
+        """
+        Parameters
+        ----------
+        subarray: SubarrayDescription
+            Description of the subarray. Provides information about the
+            camera which are useful in calibration. Also required for
+            configuring the TelescopeParameter traitlets.
+        config: traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            This is mutually exclusive with passing a ``parent``.
+        parent: ctapipe.core.Component or ctapipe.core.Tool
+            Parent of this component in the configuration hierarchy,
+            this is mutually exclusive with passing ``config``
+        """
+
+        super().__init__(subarray=subarray, config=config, parent=parent, **kwargs)
+
+        self.calib = CameraCalibrator(subarray=self.subarray, parent=self)
         self.ring_fitter = MuonRingFitter(parent=self)
-        self.intensity_fitter = MuonIntensityFitter(subarray=subarray, parent=self)
-        self.cleaning = TailcutsImageCleaner(parent=self, subarray=subarray)
-        self.writer = HDF5TableWriter(
-            self.output, "", add_prefix=True, parent=self, mode="w"
-        )
+        self.intensity_fitter = MuonIntensityFitter(subarray=self.subarray, parent=self)
+        self.cleaning = TailcutsImageCleaner(parent=self, subarray=self.subarray)
+
         self.pixels_in_tel_frame = {}
         self.field_of_view = {}
         self.pixel_widths = {}
 
         for p in ["min_pixels", "pedestal", "ratio_width", "completeness_threshold"]:
-            getattr(self, p).attach_subarray(self.source.subarray)
-
-    def start(self):
-        for event in tqdm(self.source, desc="Processing events: "):
-            self.process_array_event(event)
+            getattr(self, p).attach_subarray(self.subarray)
 
-    def process_array_event(self, event):
+    def __call__(self, event: ArrayEventContainer):
         self.calib(event)
```

Comment:
see previous comment

Reply:
Again, this is intentional and has also already been discussed before here #1353